src/ibc/lightclients/tendermint/v1/tendermint.ts

```diff
@@ -42,41 +57,44 @@ export interface ClientState {
    * the default upgrade module, upgrade_path should be []string{"upgrade",
    * "upgradedIBCState"}`
    */
+
   upgradePath: string[];
   /**
    * This flag, when set to true, will allow governance to recover a client
    * which has expired
    */
+
   allowUpdateAfterExpiry: boolean;
   /**
    * This flag, when set to true, will allow governance to unfreeze a client
    * whose chain has experienced a misbehaviour event
    */
+
   allowUpdateAfterMisbehaviour: boolean;
 }
-
 /** ConsensusState defines the consensus state from Tendermint. */
+
 export interface ConsensusState {
   /**
    * timestamp that corresponds to the block height in which the ConsensusState
    * was stored.
    */
-  timestamp?: Timestamp;
+  timestamp: Timestamp;
   /** commitment root (i.e app hash) */
-  root?: MerkleRoot;
+
+  root: MerkleRoot;
   nextValidatorsHash: Uint8Array;
 }
-
 /**
  * Misbehaviour is a wrapper over two conflicting Headers
  * that implements Misbehaviour interface expected by ICS-02
  */
+
 export interface Misbehaviour {
   clientId: string;
-  header1?: Header;
-  header2?: Header;
+  header_1: Header;
+  header_2: Header;
```

Comment:
This would break the client code using the fields. Why are `_` used here? All other fields are in camelCase, so the `_` feels a bit misplaced.

Reply:
- [ ] https://github.com/osmosis-labs/telescope/issues/66

Reply:
@webmaster128 the protos look like this https://github.com/cosmos/ibc-go/blob/c8e2ebcf2eaa6b487ca52499063c7c3db4253398/proto/ibc/lightclients/tendermint/v1/tendermint.proto#L84-L85

```proto
message Misbehaviour {
  option (gogoproto.goproto_getters) = false;

  // ClientID is deprecated
  string client_id = 1 [deprecated = true, (gogoproto.moretags) = "yaml:\"client_id\""];
  Header header_1  = 2 [(gogoproto.customname) = "Header1", (gogoproto.moretags) = "yaml:\"header_1\""];
  Header header_2  = 3 [(gogoproto.customname) = "Header2", (gogoproto.moretags) = "yaml:\"header_2\""];
}
```

Reply:
I suppose I could camelCase the `gogoproto.customname` to get the expected result. Other than that, removing the underscore I don't think is a default to the camelCase function that is used by protobuf.js

Reply:
specfically, the `camelCase` as defined in protobuf.js is this one

```js
var camelCaseRe = /_([a-z])/g;
const camelCase = function camelCase(str) {
    return str.substring(0, 1)
         + str.substring(1)
               .replace(camelCaseRe, function($0, $1) { return $1.toUpperCase(); });
}
```

I'm guessing we may want to consider if we see a pattern like `/[a-zA-Z0-9]+_[0-9]+%$/`, we could remove the underscore?

Reply:
UPDATE: I decided to use the assumption that if there is an underscore with only a number after it, then the underscore should be removed. This is only in the case where it ends with numbers only, if there are any words we leave it alone.